osquery/sql/sqlite_collation.cpp

```diff
@@ -0,0 +1,257 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <string>
+
+#include <sqlite3.h>
+
+namespace osquery {
+/**
+ * @brief Compares the position of epoch delimiter between two versions.
+ * Return difference of epoch segment length if both versions have epoch.
+ * Return 1 if left version has epoch but not the right.
+ * Return -1 if right version has epoch but not the left.
+ * Return 0 if epoch doesn't exist in either version.
+ */
+static int compareEpoch(int lLen,
+                        const char* lVer,
+                        int rLen,
+                        const char* rVer) {
+  auto lEpoch = strcspn(lVer, ":");
+  auto rEpoch = strcspn(rVer, ":");
+
+  if (lEpoch != lLen && rEpoch != rLen) {
```

Comment:
Why does this compare `lEpoch != lLen` ?

Reply:
If none of the match characters are found in the strcspn, it  returns the length of the string. I'm using that to check if both versions have epoch by `epoch != len`.

Reply:
So... If there's no epoch, then `strcspn` returns the length of the string, so you're effectively checking for that? I don't know if that's something c++ folks would just know, but this idiom is really confusing. A comment to this effect would help. `If there is no match, strcspn returns the length of the string. Thus, we compare the length here` or something

Reply:
Using string_view with find now, and comparing that to ver.npos, so that should hopefully be clearer now.